.github/workflows/deploy-infra.yml

```diff
@@ -0,0 +1,48 @@
+name: DeployInfra
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'infra/**'
+jobs:
+  deploy:
+    name: DeployInfra
+    runs-on: ubuntu-latest
+    environment: di-ipv-cri-address-infra-build
+    timeout-minutes: 15
+    env:
+      AWS_REGION: eu-west-2
+      ENVIRONMENT: build
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup SAM
+        uses: aws-actions/setup-sam@v1
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.GH_ACTIONS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+      
+      - name: SAM build
+        run: sam build -t infra/template.yaml --config-env ${{ env.ENVIRONMENT }}
```

Comment:
I think we can drop the `--config-env` flag for this `sam build` and the `sam package` below.